server/blob/blob-objectstorage/src/main/java/org/apache/james/blob/objectstorage/swift/SwiftKeystone3ObjectStorage.java

```diff
@@ -0,0 +1,243 @@
+            this.identity = identity;
+            this.region = region;
+            this.credentials = credentials;
+            this.project = project;
+            this.domainId = domainId;
+        }
+
+        public URI getEndpoint() {
+            return endpoint;
+        }
+
+        public IdentityV3 getIdentity() {
+            return identity;
+        }
+
+        public Credentials getCredentials() {
+            return credentials;
+        }
+
+        public Properties getOverrides() {
+            Properties properties = new Properties();
+            properties.setProperty(KeystoneProperties.KEYSTONE_VERSION, "3");
+            OptionalUtils.or(
+                project.map(project -> setScope(project)),
+                domainId.map(domain -> setScope(domain))
+            ).ifPresent(scopeProps ->
+                properties.putAll(scopeProps)
+            );
+            return properties;
+        }
+
+        private Properties setScope(DomainId domainId) {
+            Properties properties = new Properties();
+            properties.setProperty(KeystoneProperties.SCOPE, domainId.asString());
+            return properties;
+        }
+
+        private Properties setScope(Project project) {
+            Properties properties = new Properties();
+            properties.setProperty(KeystoneProperties.SCOPE, project.name().asString());
+            project.domainName()
+                .map(domain -> Pair.of(KeystoneProperties.PROJECT_DOMAIN_NAME, domain));
+            OptionalUtils.or(
+                project.domainName()
+                    .map(domain -> Pair.of(KeystoneProperties.PROJECT_DOMAIN_NAME, domain.value())),
+                project.domainId()
+                    .map(domain -> Pair.of(KeystoneProperties.PROJECT_DOMAIN_ID, domain.value()))
+            ).ifPresent(pair ->
+                properties.setProperty(pair.getKey(), pair.getValue())
+            );
+
+            return properties;
+        }
+
+        public Optional<Region> getRegion() {
+            return region;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) {
+                return true;
+            }
+            if (o == null || getClass() != o.getClass()) {
+                return false;
+            }
+            Configuration that = (Configuration) o;
+            return Objects.equal(endpoint, that.endpoint) &&
+                Objects.equal(identity, that.identity) &&
+                Objects.equal(region, that.region) &&
+                Objects.equal(credentials, that.credentials) &&
+                Objects.equal(project, that.project) &&
+                Objects.equal(domainId, that.domainId);
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hashCode(endpoint, identity, region, credentials, project, domainId);
+        }
+
+        @Override
+        public String toString() {
+            return "Configuration{" +
```

Comment:
Why not MoreObjects?

Reply:
because I forgot to change the intellij generated code :)